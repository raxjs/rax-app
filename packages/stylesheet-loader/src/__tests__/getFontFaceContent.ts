import { parse, getFontFaceContent } from '../index';
import * as css from 'css';
import * as loaderUtils from 'loader-utils';

describe('getFontFaceContent', () => {
  it('should convert `@font-face` correctly when use double quotes', () => {
    const font = `@font-face {
  font-family: "Open Sans";
  src: url("/fonts/OpenSans-Regular-webfont.woff2") format("woff2"),
        url("/fonts/OpenSans-Regular-webfont.woff") format("woff");
}`;

    // getOptions can return null if no query passed.
    const parsedQuery = loaderUtils.getOptions(this) || {};

    // Compatible with string true.
    if (parsedQuery.log === 'true') {
      parsedQuery.log = true;
    }

    const { stylesheet } = css.parse(font);
    const { fontFaceRules } = parse(parsedQuery, stylesheet);

    expect(getFontFaceContent(fontFaceRules)).toMatchSnapshot();
  });

  it('should convert `@font-face` correctly when use single quotes', () => {
    const font = `@font-face {
  font-family: 'Open Sans';
  src: url('/fonts/OpenSans-Regular-webfont.woff2') format('woff2'),
        url('/fonts/OpenSans-Regular-webfont.woff') format('woff');
}`;

    // getOptions can return null if no query passed.
    const parsedQuery = loaderUtils.getOptions(this) || {};

    // Compatible with string true.
    if (parsedQuery.log === 'true') {
      parsedQuery.log = true;
    }

    const { stylesheet } = css.parse(font);
    const { fontFaceRules } = parse(parsedQuery, stylesheet);

    expect(getFontFaceContent(fontFaceRules)).toMatchSnapshot();
  });
});
