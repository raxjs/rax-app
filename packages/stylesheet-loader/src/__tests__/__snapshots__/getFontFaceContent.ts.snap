// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFontFaceContent should convert \`@font-face\` correctly when use double quotes 1`] = `
"
  if (typeof FontFace === 'function') {
    
    var fontFace0 = new FontFace('Open Sans', 'url(\\"/fonts/OpenSans-Regular-webfont.woff2\\") format(\\"woff2\\"),
        url(\\"/fonts/OpenSans-Regular-webfont.woff\\") format(\\"woff\\")');
    document.fonts.add(fontFace0);
    
  }
    "
`;

exports[`getFontFaceContent should convert \`@font-face\` correctly when use single quotes 1`] = `
"
  if (typeof FontFace === 'function') {
    
    var fontFace0 = new FontFace('Open Sans', 'url(\\"/fonts/OpenSans-Regular-webfont.woff2\\") format(\\"woff2\\"),
        url(\\"/fonts/OpenSans-Regular-webfont.woff\\") format(\\"woff\\")');
    document.fonts.add(fontFace0);
    
  }
    "
`;
